plugins/kubernetes/app/assets/javascripts/kubernetes/directives/kubernetes_deploy_progress_directive.js

```diff
@@ -0,0 +1,64 @@
+samson.directive('deployProgressWidget', function() {
+  return {
+    restrict: 'E',
+    templateUrl: 'kubernetes/deploy_progress_widget.tmpl.html',
+    scope: {
+      role: '=',
+      deployGroup: '='
+    },
+    controller: function($scope) {
+
+      $scope.currentRelease = function(release) {
+        var current = $scope.lastRelease();
+        return release.id == current.id;
+      };
+
+      $scope.progress = function(release) {
+        return ((release.live_replicas * 100) / release.target_replicas) + '%';
+      };
+
+      $scope.deployCompleted = function() {
+        return _.every($scope.deployGroup.releases, function(release) {
+          return $scope.currentRelease(release) ? release.live_replicas == release.target_replicas : release.live_replicas == 0;
```

Comment:
Oh, you're right. I created that method afterwards. Going to refactor it, thanks for the heads up.